go/vt/vttablet/tabletserver/vstreamer/vstreamer.go

```diff
@@ -268,24 +269,36 @@ func (vs *vstreamer) parseEvents(ctx context.Context, events <-chan mysql.Binlog
 	}
 
 	// Main loop: calls bufferAndTransmit as events arrive.
-	timer := time.NewTimer(HeartbeatTime)
-	defer timer.Stop()
+	hbTimer := time.NewTimer(HeartbeatTime)
+	defer hbTimer.Stop()
+
+	injectHeartbeat := func(throttled bool) error {
+		now := time.Now().UnixNano()
+		err := bufferAndTransmit(&binlogdatapb.VEvent{
+			Type:        binlogdatapb.VEventType_HEARTBEAT,
+			Timestamp:   now / 1e9,
+			CurrentTime: now,
+			Throttled:   throttled,
```

Comment:
I'm curious why we get a nanosecond precision timestamp only to turn it back into a second precision timestamp for the protobuf message?

Reply:
I'm not sure, this block of code is moved from elsewhere and otherwise untouched.

Reply:
OK, let's assume there was some good reason. 😄